I changed "pragma GCC diagnostic to pragma clang diagnostic."
**make failed:**

ClickHouse/dbms/src/Common/Collator.cpp:13:0: error: ignoring #pragma clang diagnostic [-Werror=unknown-pragmas]
     #pragma clang diagnostic ignored "-Wunused-private-field"

cc1plus: all warnings being treated as errors
dbms/CMakeFiles/dbms.dir/build.make:5942: recipe for target 'dbms/CMakeFiles/dbms.dir/src/Common/Collator.cpp.o' failed
make[2]: *** [dbms/CMakeFiles/dbms.dir/src/Common/Collator.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
^Cmake[2]: *** Deleting file 'dbms/CMakeFiles/dbms.dir/src/Dictionaries/ComplexKeyCacheDictionary.cpp.o'
dbms/CMakeFiles/dbms.dir/build.make:3878: recipe for target 'dbms/CMakeFiles/dbms.dir/src/Dictionaries/CacheDictionary.cpp.o' failed
make[2]: *** [dbms/CMakeFiles/dbms.dir/src/Dictionaries/CacheDictionary.cpp.o] Interrupt
dbms/CMakeFiles/dbms.dir/build.make:3926: recipe for target 'dbms/CMakeFiles/dbms.dir/src/Dictionaries/ComplexKeyCacheDictionary.cpp.o' failed
make[2]: *** [dbms/CMakeFiles/dbms.dir/src/Dictionaries/ComplexKeyCacheDictionary.cpp.o] Interrupt
CMakeFiles/Makefile2:4039: recipe for target 'dbms/CMakeFiles/dbms.dir/all' failed
make[1]: *** [dbms/CMakeFiles/dbms.dir/all] Interrupt
Makefile:160: recipe for target 'all' failed
make: *** [all] Interrupt

--- a/dbms/src/Common/Collator.cpp
+++ b/dbms/src/Common/Collator.cpp
@@ -9,6 +9,11 @@
     #endif
     #include <unicode/ucol.h>
     #pragma GCC diagnostic pop
+#else
+    #if __clang__
+        #pragma clang diagnostic push
+        #pragma clang diagnostic ignored "-Wunused-private-field"
+    #endif
 #endif
 
 #include <Common/Exception.h>

